### ISSUE TITLE
Source code has no docstrings

### ISSUE THREAD

**Issue by Granitosaurus** [microsoft/playwright-python#170]
I'm trying to learn some of playwrights internals but it's really hard to follow when none of the code has any documentation. Is it stored somewhere else or the project just doesn't have anything implemented just yet?

**Comment by pavelfeldman** [comment]
@Granitosaurus : are you using an IDE or looking for the docstrings in the source? Both [Sync API](https://github.com/microsoft/playwright-python/blob/master/playwright/sync_api.py) and [Async API](https://github.com/microsoft/playwright-python/blob/master/playwright/async_api.py) have a lot of docs. The project is implemented in full. The API is quite large, so it might be easier to look at the [API reference](https://playwright.dev/#version=v1.3.0&path=docs%2Fapi.md).

**Comment by Granitosaurus** [comment]
Yeah I'm using IDE to browse the source code. The sync/async `BrowserType` is well documented indeed, my IDE just couldn't find the correct reference!
  
However most helpers and individual components don't have any docstrings and are a bit hard to follow.

**Comment by pavelfeldman** [comment]
All the individual components are an implementation detail of Playwright, only sync/async apis are exposed to the user.

**Comment by mxschmitt** [comment]
There are currently no plans to add docstrings to the internal methods. If you are interested in contributing to Playwright itself, most of the magic happens upstream: https://github.com/microsoft/playwright
